iceoryx_posh/source/popo/building_blocks/chunk_queue.cpp

```diff
@@ -35,31 +35,35 @@ ChunkQueue::MemberType_t* ChunkQueue::getMembers() noexcept
     return m_chunkQueueDataPtr;
 }
 
-bool ChunkQueue::push(mepoo::SharedChunk chunk) noexcept
+cxx::expected<ChunkQueueError> ChunkQueue::push(mepoo::SharedChunk chunk) noexcept
 {
     ChunkQueueData::ChunkTuple chunkTupleIn(chunk.releaseWithRelativePtr());
 
-    return !getMembers()
-                ->m_queue.push(chunkTupleIn)
-                .on_success(
-                    [this](cxx::expected<cxx::optional<ChunkQueueData::ChunkTuple>, cxx::VariantQueueError>& retVal) {
-                        // drop the chunk if one is returned by a safe overflow
-                        if (*retVal)
-                        {
-                            auto chunkTupleOut = **retVal;
-                            auto chunkManagement = iox::relative_ptr<mepoo::ChunkManagement>(
-                                chunkTupleOut.m_chunkOffset, chunkTupleOut.m_segmentId);
-                            auto chunk = mepoo::SharedChunk(chunkManagement);
-                        }
-
-                        if (getMembers()->m_semaphoreAttached.load(std::memory_order_acquire)
-                            && getMembers()->m_semaphore)
-                        {
-                            std::cout << "fubar" << std::endl;
-                            getMembers()->m_semaphore->post();
-                        }
-                    })
-                .has_error();
+    auto pushRet = getMembers()->m_queue.push(chunkTupleIn);
+
+    if (pushRet.has_error())
+    {
+        return cxx::error<ChunkQueueError>(ChunkQueueError::QUEUE_OVERFLOW);
+    }
+    else
+    {
+        // drop the chunk if one is returned by a safe overflow
+        if ((*pushRet).has_value())

```

Comment:
hmmm, for my not. (pushRet.get_value()).has_value()? But yes optional and expected allow a lot of different flavors. I prefer has_value() and has_error() and not the bool operators, for better readability. But I like * operator instead of get_value().

Reply:
I have no hard feelings about this. My eye just notices `.get_value()` easier than `*` ;-)